### PR TITLE
Updated MediaStream to the 2016 specs

### PR DESCRIPTION
**Pull request by mvh77**
New PR for the updated MediaStream, with proper branch and all!

### CODE REVIEW

File: src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala [hunk 1]
```diff
@@ -314,13 +327,55 @@ trait MediaStreamTrack extends EventTarget {
...
+  var facingMode: String = js.native
+  var volume: Double = js.native
+  var sampleRate: Double = js.native
+  var sampleSize: Double = js.native
+  var echoCancellation: Boolean = js.native
+  var latency: Double = js.native
+  var channelCount: Double = js.native
+  var deviceId: String = js.native
+  var groupId: String = js.native
 }
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala]: I suggest using an `@ScalaJSDefined` trait with `= js.undefined` members. This is the new recommended style for "configuration objects".

File: src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala [hunk 2]
```diff
@@ -314,13 +327,55 @@ trait MediaStreamTrack extends EventTarget {
...
+    volume.foreach(result.volume = _)
+    sampleRate.foreach(result.sampleRate = _)
+    sampleSize.foreach(result.sampleSize = _)
+    echoCancellation.foreach(result.echoCancellation = _)
+    latency.foreach(result.latency = _)
+    channelCount.foreach(result.channelCount = _)
+    deviceId.foreach(result.deviceId = _)
+    groupId.foreach(result.groupId = _)
+    result.asInstanceOf[MediaTrackConstraintSet]
+  }
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala]: With `= js.undefined`, this separate constructor is not needed.

File: src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala [hunk 3]
```diff
@@ -576,3 +609,78 @@ trait NavigatorMediaStream extends js.Object {
...
+  var facingMode: Boolean = js.native
+  var volume: Boolean = js.native
+  var sampleRate: Boolean = js.native
+  var sampleSize: Boolean = js.native
+  var echoCancellation: Boolean = js.native
+  var latency: Boolean = js.native
+  var channelCount: Boolean = js.native
+  var deviceId: Boolean = js.native
+  var groupId: Boolean = js.native
+}
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/mediastream/MediaStream.scala]: Same here: use `= js.undefined` members.

**Comment by sjrd** [comment]
Thanks. Could you squash the 3 commits, please? (Keeping only the first commit message) Then it's good to go.

**Comment by mvh77** [comment]
Could this be merged?

**Pull request merged**